components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,412 @@
+		},
+	}
+	registry.RegisterComponent(motor.Subtype, modelName, _motor)
+
+	config.RegisterComponentAttributeMapConverter(
+		motor.SubtypeName,
+		modelName,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf Config
+			decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Squash: true, Result: &conf})
+			if err != nil {
+				return nil, err
+			}
+			if err := decoder.Decode(attributes); err != nil {
+				return nil, err
+			}
+			return &conf, nil
+		},
+		&Config{},
+	)
+}
+
+// NewMotor returns a Sabertooth driven motor.
+func NewMotor(ctx context.Context, c *Config, logger golog.Logger) (motor.LocalMotor, error) {
+	if c.SerialDevice == "" {
+		devs := usb.Search(usbFilter, func(vendorID, productID int) bool {
+			if vendorID == 0x403 && productID == 0x6001 {
+				return true
+			}
+			return false
+		})
+
+		if len(devs) > 0 {
+			c.SerialDevice = devs[0].Path
+		} else {
+			return nil, errors.New("couldn't find Sabertooth serial connection")
+		}
+	}
+
+	globalMu.Lock()
+	ctrl, ok := controllers[c.SerialDevice]
+	if !ok {
+		newCtrl, err := newController(c, logger)
+		if err != nil {
+			return nil, err
+		}
+		controllers[c.SerialDevice] = newCtrl
+		ctrl = newCtrl
+	}
+	globalMu.Unlock()
+
+	ctrl.mu.Lock()
+	defer ctrl.mu.Unlock()
+
+	// is on a known/supported amplifier only when map entry exists
+	claimed, ok := ctrl.activeAxes[c.Channel]
+	if !ok {
+		return nil, fmt.Errorf("invalid Sabertooth motor axis: %d", c.Channel)
+	}
+	if claimed {
+		return nil, fmt.Errorf("axis %d is already in use", c.Channel)
+	}
+	ctrl.activeAxes[c.Channel] = true
+
+	m := &Motor{
+		c:       ctrl,
+		Channel: c.Channel,
+	}
+
+	if err := m.configure(c); err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+// Close stops the motor and marks the axis inactive.
+func (m *Motor) Close() {
+	m.c.mu.Lock()
+	active := m.c.activeAxes[m.Channel]
+	m.c.mu.Unlock()
+	if !active {
+		return
+	}
+	err := m.Stop(context.Background(), nil)
+	if err != nil {
+		m.c.logger.Error(err)
+	}
+
+	m.c.mu.Lock()
+	defer m.c.mu.Unlock()
+	m.c.activeAxes[m.Channel] = false
+	for _, active = range m.c.activeAxes {
+		if active {
+			return
+		}
+	}
+	if m.c.port != nil {
+		err = m.c.port.Close()
+		if err != nil {
+			m.c.logger.Error(err)
+		}
+	}
+	globalMu.Lock()
+	defer globalMu.Unlock()
+	delete(controllers, m.c.serialDevice)
+}
+
+func newController(c *Config, logger golog.Logger) (*controller, error) {
+	if c.Address < 128 || c.Address > 135 {
+		return nil, errors.New("address is out of range. valid addresses are 127 to 135")
+	}
+	ctrl := new(controller)
+	ctrl.activeAxes = make(map[int]bool)
+	ctrl.serialDevice = c.SerialDevice
+	ctrl.logger = logger
+	ctrl.address = c.Address
+
+	if c.TestChan != nil {
+		ctrl.testChan = c.TestChan
+	} else {
+		serialOptions := serial.OpenOptions{
+			PortName:          c.SerialDevice,
+			BaudRate:          9600,
```

Comment:
Will baudrate ever be configurable?

Reply:
It is possible to change it, that's why `common.go` has the `baud2400`, `baud9600`, etc. Is there a way to express the desire to change the baudrate in viam?

Reply:
same as you've used the config for serial device for port name, you can have a c.BaudRate field and you can set it through config fields in app. `movementsensor/gps-nmea` has an example of this as I recall.

You can also check if it's not set, and se a default (9600) and log  it as info that way.